rewrite-java/src/main/java/org/openrewrite/java/tree/J.java

```diff
@@ -363,6 +367,58 @@ public CoordinateBuilder.Expression getCoordinates() {
         }
     }
 
+    @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+    @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
+    @Data
+    final class ArrayTypeTree implements J, TypeTree, Expression {
+        @With
+        @EqualsAndHashCode.Include
+        UUID id;
+
+        @With
+        Space prefix;
+
+        @With
+        Markers markers;
+
+        @With
+        TypeTree elementType;
+
+        @Nullable
+        @With
+        List<J.Annotation> annotations;
+
+        @With
+        JRightPadded<Space> dimension;
```

Comment:
That does make more sense, I used `JRightPadded<Space> dimension` since the other dimensions are modeled that way, but it doesn't mean it's correct. I'll updated it tomorrow.